spec/rubocop/cop/style/guard_clause_spec.rb

```diff
@@ -323,6 +323,33 @@ def func
     RUBY
   end
 
+  it 'registers an offense when using lvar as an argument of raise in `else` branch' do
+    expect_offense(<<~RUBY)
+      def func
+      rescue => e
+        unless condition
+        ^^^^^^ Use a guard clause (`raise e if condition`) instead of wrapping the code inside a conditional expression.
+          retry
+        else
+          raise e
+        end
+      end
```

Comment:
Can you update the following minimum repro case?

```ruby
      if condition
      ^^ Use a guard clause (`raise e unless condition`) instead of wrapping the code inside a conditional expression.
        do_something
      else
        raise e
      end
```